Add push rule for `.m.rule.reaction`

oppiliappan opened the pull request:
- resides behind the feature flag `unstable-pre-spec`
 - fixes #371 

I would like to know if the doc comment is appropriate.

oppiliappan committed:
Add push rule for `.m.rule.reaction`

 - resides behind the feature flag `unstable-pre-spec`

jplatte commented on ruma/Cargo.toml:
```diff
@@ -57,6 +57,7 @@ unstable-pre-spec = [
     "ruma-client-api/unstable-pre-spec",
     "ruma-events/unstable-pre-spec",
     "ruma-federation-api/unstable-pre-spec",
+    "ruma-common/unstable-pre-spec",
```
There are dependencies to ruma-common in other crates that expose an `unstable-pre-spec` feature. Please also add `ruma-common/unstable-pre-spec` to their unstable-pre-spec feature dependencies.

oppiliappan replied:
I have added it to
 - ruma-events
 - ruma-federation-api
 - ruma-client-api

jplatte commented on ruma-common/src/push/predefined.rs:
```diff
@@ -146,6 +146,18 @@ impl OverridePushRule {
         })
     }
+
+    /// MSC2677: Annotations and Reactions
```
It's good to mention the MSC, but the docs should also explain what this rule does (like the docs for the other rules).

jplatte reviewed (commented):
About the actual code: The new method should also be called in `server_default` near the start of the file (conditionally under the `unstable-pre-spec` feature flag).

oppiliappan committed:
Add ruma-common to unstable-pre-spec feature flags

 - include `reaction` method in `server_default`
 - document `reaction` method in detail

jplatte reviewed (approved):
Docs could still be expanded, but with this being an unstable feature I don't think it's very important right now.

jplatte commented:
Thanks!
